### DISCUSSION TITLE
Tree generator error

### DISCUSSION THREAD

**Discussion by qinqinqinj** [UMEP-dev/UMEP#561] [Q&A]
I'm using version 3.28 of QGIS.
I have used a point vector layer to represent the position of the tree, where the attribute table contains the 4 attributes required; and also used DSM and DEM layers, wanting to generate TDSM and CDSM. but the generated layers TDSM (TDSMA) and CDSM (CDSMA) show zero bands, and I don't know why this is happening.
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/801edb0e-f9fd-48b5-92dc-ed036c88803d)
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/7ec70e99-d664-4208-85ce-3044f8039671)
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/f4fcb915-174a-461e-97a1-0c507283b90b)
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/d35023e2-b6b0-42e7-80ad-f7093e1711ed)
I tried the case provided on the UMEP website to try it out and it generates trees correctly.
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/31e8a7f7-daab-4d38-83d8-8ff4f80f7c3b)
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/85666628-3c26-441b-8678-f6b1b6092c12)
I referred to the previous answer and realized that the difference between my own file and the case file is that the coordinate reference system (CRS) is different, is there a problem here? How to solve it?
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/b349aa96-b718-49b2-b884-38921883410a)
(The above is the CRS for my file)
![image](https://github.com/UMEP-dev/UMEP/assets/147718158/54eaa54d-2636-45c4-b6f8-ad91636b930c)
(The above is the CRS of the reference document)
Here are the files I used：
[Documents utilized.zip](https://github.com/UMEP-dev/UMEP/files/13062846/Documents.utilized.zip)
Please answer the questions!

**Comment by biglimp** [comment]
First you can do is to reproject all your data into the same coordinate reference system (CRS). Algorithms for this is found in the Processing Toolbox in QGIS. Also, use a CRS that is represented in meters, e.g. EPSG 26918.

**Reply by qinqinqinj** [reply]
Thank you very much for your reply, my diagram should not have these problems. I'll leave the problem with my graph here for your reference. The problem with my diagram is that the pixels in the overall diagram are too large and the pixels generated by the tree are too small, so it looks black on the diagram, I don't know why the bands show up as zero, but the CDSM and TDSM are generated correctly, it's just that the pixels show up too small in the overall image to be observed, which is possible to observe if you use other image software.
